src/librados/IoCtxImpl.cc

```diff
@@ -1962,7 +1962,9 @@ librados::IoCtxImpl::C_aio_Complete::C_aio_Complete(AioCompletionImpl *_c)
 void librados::IoCtxImpl::C_aio_Complete::finish(int r)
 {
   c->lock.Lock();
-  c->rval = r;
+  // Leave an existing rval unless r != 0
+  if (r)
```

Comment:
@jdurgin @neha-ojha This is the only part of this pull request that gives me concern.  Hopefully, a rados run will confirm that this doesn't break anything.

Reply:
I don't think it should be a problem, tests will tell though